index.js

```diff
@@ -0,0 +1,36 @@
+const $taskInput = document.getElementById('taskInput');
+const $taskList = document.getElementById('taskList');
+
+const handleKeydownTaskInput = (event) => {
+  const taskText = $taskInput.value.trim();
+
+  if (event.key !== 'Enter' || taskText === '') {
+    return;
+  }
```

Comment:
`taskText === ''`를 `!task`로 작성해도 문제없이 동작할 것 같은데요. 두 개의 표현식은 어떤 차이가 있을까요?

키워드
- falsy

```suggestion
  if (event.key !== 'Enter' || !taskText) {
    return;
  }
```

Reply:
- taskText === '': taskText가 정확히 빈 문자열인 경우에만 참이 됩니다.

- !taskText: taskText가 falsy한 값이면(빈 문자열이거나 다른 falsy한 값이면) 참이 됩니다.
 taskText가 빈 문자열 뿐만 아니라 null, undefined, 숫자 0, false와 같은 값들도 모두 참으로 평가됩니다.

Reply:
맞아요~ 잘 정리해줬어요.

현재와 같이 input의 입력값을 검증하는 경우에는 둘중에 어떤걸 써도 괜찮지만 저는 개인적으로 falsy값을 사용하는 것 대신에 명시적으로 적어주는게 좋다고 생각해요. 혹시나 falsy값중에 미처 파악하지 못한 유효한 값(0, null)이 있을수도 있는데 그러면 문제가 생길 수 있어요. 이러한 문제로 버그가 발생하는 경우도 있구요. 그리고 코드를 읽을때도 조금 더 명확해지는 느낌이 있어요.

`if(!taskText)`는 taskText가 falsy값이면, ~한다.
`if(taskText === '')`는 taskText가 빈 문자열이면, ~한다.

이렇게 어떠한 경우에 해당하는건지 더욱 좁은 범위로 좁힐 수 있어서 코드를 처음 읽는다고 해도 이해하기 좋아요.

추가적으로 조건식을 리팩토링하는 방법중에 변수로 추출하는 방법이 있어요.
지금정도의 조건식은 코드를 그대로 읽어도 이해하기 쉽지만, 만약에 조건식이 복잡해진다면 사용하기 좋은 방법이에요.

아래와 같이 if문에 들어가는 조건을 그대로 변수에 담고, if문에는 변수를 넣어주는 방식이에요.
이렇게 하면 코드를 읽을 때, 실제로 어떤 경우가 예외 케이스인지는 신경쓰지 않고, `잘못된 경우에는 함수의 실행을 종료한다.` 이렇게 코드를 이해할 수 있어요~

```suggestion
  const invalidCase = event.key !== 'Enter' || taskText === '';
  
  if (inValidCase) {
    return;
  }
```

또 토스에서는 금융 관련된 것을 다루다보니 조건식이 복잡한 경우가 많아서(ex. 약관에 동의한 경우, 개인정보처리방침에 동의한 경우등) 한글 변수명을 많이 쓰고 있다고 해요~
다만 실제로 한글 변수명을 회사에서 쓰려면 팀원들간의 동의가 필요하고 납득하지 못하는 경우도 많아서 미리 얘기하고 합의하는 게 중요해요. 개인적으로 개발할때는 복잡한 조건의 경우에는 한글 변수명을 쓰고 있어요~

```suggestion
  const 유효하지_않으면 = event.key !== 'Enter' || taskText === '';
  
  if (유효하지_않으면) {
    return;
  }
```

Reply:
우와우와 글이 진짜 술술 읽히고 
저도 써보고 싶어서 복잡한 조건식을 만났으면 하고 기다려져염 ㅎㅎ